src/compiler/types.ts

```diff
@@ -6053,7 +6053,7 @@ export const enum TypeFlags {
     /** @internal */
     Intrinsic = Any | Unknown | String | Number | BigInt | Boolean | BooleanLiteral | ESSymbol | Void | Undefined | Null | Never | NonPrimitive,
     /** @internal */
-    Primitive = String | Number | BigInt | Boolean | Enum | EnumLiteral | ESSymbol | Void | Undefined | Null | Literal | UniqueESSymbol | TemplateLiteral,
+    Primitive = String | Number | BigInt | Boolean | Enum | EnumLiteral | ESSymbol | Void | Undefined | Null | Literal | UniqueESSymbol | TemplateLiteral | StringMapping,

```

Comment:
I wanted to reuse `StringLike` here but that would require me to reorder those enum members and I wasn't sure if you'd like that

Reply:
I think consolidating these into the Likes would be preferable.

Reply:
done